eng/ci/templates/official/jobs/setup-benchmark-agents.yml

```diff
@@ -1,29 +1,51 @@
 parameters:
 - name: functionAppName
   type: string
-- name: agentName
+- name: os
   type: string
+  default: Windows
+ 
 jobs:
-- job: ${{ parameters.functionAppName }}
+- job: ${{ parameters.functionAppName }}_${{ parameters.os }}
+
+  ${{ if eq(parameters.os, 'Linux') }}:
+    pool:
+      name: 1es-pool-azfunc-benchmarking-large
+      image: 1es-ubuntu-22.04-benchmark-runner-vanilla
+      os: linux 
+  ${{ else }}:
+    pool:
+      name: 1es-pool-azfunc-benchmarking-large
+      image: 1es-windows-2022-benchmark-runner-vanilla
+      os: windows
 
   variables:
-    agentName: ${{ parameters.agentName }}
+    agentId: ${{ parameters.functionAppName }}${{ parameters.os }}
     functionApp: ${{ parameters.functionAppName }}
     functionAppOutputPath: $(Build.ArtifactStagingDirectory)/FunctionApps/$(functionApp)
 
   steps:
 
   - template: /eng/ci/templates/install-dotnet.yml@self
 
+  # Adds the Ubuntu 20.04 security repository to Ubuntu 22.04 and installs libssl1.1, which is unavailable by default.
+  - script: |
+      echo "deb http://security.ubuntu.com/ubuntu focal-security main" | sudo tee /etc/apt/sources.list.d/focal-security.list
+      sudo apt update && sudo apt install -y libssl1.1
+    displayName: "Install libssl1.1 on Ubuntu 22.04"
+    condition: eq(variables['Agent.OS'], 'Linux')
+
   - script: dotnet tool install -g Microsoft.Crank.Agent --version "0.2.0-*"
     displayName: Install Microsoft.Crank.Agent tool
 
   - pwsh: Start-Process powershell -ArgumentList '-NoExit', '-Command', 'crank-agent'
```

Comment:
Consider using templating to if/else these two steps:

``` yaml
- ${{ if eq(parameters.os, 'Windows') }}:
  - pwsh: ... # windows step

- ${{ else }}:
  - script: ... # linux step
```

The difference between this approach and using `condition` is that this will occur during templating. The resulting yaml will only include the appropriate step for the OS. The `conditon` approach is lazily evaluated at the time of step execution. So, the templating approach is more consistent (no mid-job variable changes will impact the decision!) and also gives a cleaner pipeline view, as there won't be the skipped steps (they won't even show up!).

I always use templating when possible. Only using `condition` when I absolutely must dynamically decide run or not based on a previous step.

Reply:
Good call. Switched to this approach. Was doing similar thing for the pool already.